Lib/logging/config.py

```diff
@@ -361,7 +361,7 @@ class BaseConfigurator(object):
 
     WORD_PATTERN = re.compile(r'^\s*(\w+)\s*')
     DOT_PATTERN = re.compile(r'^\.\s*(\w+)\s*')
-    INDEX_PATTERN = re.compile(r'^\[\s*(\w+)\s*\]\s*')
+    INDEX_PATTERN = re.compile(r'^\[(.*?)\]\s*')
```

Comment:
Maybe allowing leading spaces would be confusing?

Reply:
I believe yes.
However, given there is no clear specification so far, we can leave the specifics to the end user.